odoo/models.py

```diff
@@ -4088,7 +4088,9 @@ def _parent_store_update(self):
         """
         cr.execute(query.format(self._table), [prefix, tuple(self.ids)])
         modified_ids = {row[0] for row in cr.fetchall()}
-        self.browse(modified_ids).modified(['parent_path'])
+        modified_records = self.browse(modified_ids)
```

Comment:
A suggestion to fix the cache and avoid a future extra query
```python
        query = """
            UPDATE {0} child
            SET parent_path = concat(%s, substr(child.parent_path,
                    length(node.parent_path) - length(node.id || '/') + 1))
            FROM {0} node
            WHERE node.id IN %s
            AND child.parent_path LIKE concat(node.parent_path, '%%')
            RETURNING child.id, child.parent_path
        """
        cr.execute(query.format(self._table), [prefix, tuple(self.ids)])
        modified_ids, parent_paths = zip(*[(row[0], row[1]) for row in cr.fetchall()])
        records = self.browse(modified_ids)
        self.env.cache.update(records, self._fields[self._parent_name], parent_paths)
        records.modified(['parent_path'])
```

Reply:
Wouldn't it be simpler to make a dict with the result of the query?
```py
updates = dict(cr.fetchall())    # {id: parent_path}
records = self.browse(updates.keys())
self.env.cache.update(records, type(self).parent_path, updates.values())
records.modified(['parent_path'])
```
BTW your cache update is incorrect, since it updates `parent_id` !